src/main/java/recipeapplication/database/RecipeFirebaseCrud.java

```diff
@@ -22,21 +26,73 @@ public RecipeFirebaseCrud(Firestore firestore,
 
 	@Override
 	public void create(Recipe data) {
-		
-		this.firestore.collection(collection).document(data.getId()).set(data);
-		
+		System.out.println("In recipe creation section");
+
+		try {
+			if (this.firestore.collection(collection).document(data.getId()).get().get().exists() == false)  // Check if document already exists, to not allow overwriting it with create.
+				this.firestore.collection(collection).document(data.getId()).set(data);
+			else
+				throw new Exception("Failed to create, Recipe with this ID already exists!");
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		} catch (ExecutionException e) {
+			e.printStackTrace();
+		} catch (Exception e) {
+			System.out.println(e.getMessage());
+		}
 	}
 
 	@Override
-	public void read(String document) {
-		// TODO Auto-generated method stub
-		
+	public Recipe read(String document) {
+		System.out.println("In recipe reading section");
+		try {
+			DocumentSnapshot docSnap = this.firestore.collection(collection).document(document).get().get();
+			if (docSnap.exists() == false)
+				throw new Exception("Failed to read, recipe doesn't exist!");
+			else {
+				Map<String, Object> data = this.firestore.collection(collection).document(document).get().get().getData();
+
+				Recipe recipe = new Recipe();
+				recipe.setId(data.get("id").toString());
+				recipe.setCategory(data.get("category").toString());
+				recipe.setIngridiant(((List<String>) data.get("ingridiant")));
+				recipe.setPreperation(data.get("preperation").toString());
+				if (data.get("recipeImageId") != null)
+					recipe.setRecipeImageId(data.get("recipeImageId").toString());
+				return recipe;
+			}
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+			return null;
+		} catch (ExecutionException e) {
+			e.printStackTrace();
+			return null;
+		} catch (Exception e) {
+			System.out.println(e.getMessage());
+			return null;
+		}
+
 	}
 
 	@Override
 	public void update(Recipe data) {
-		// TODO Auto-generated method stub
-		
+		System.out.println("In recipe updating section");
+
+		try {
+			DocumentSnapshot docSnap = this.firestore.collection(collection).document(data.getId()).get().get();
+			if(docSnap.exists() && docSnap.get("id").toString().equalsIgnoreCase(data.getId())) {
+				this.firestore.collection(collection).document(data.getId()).set(data);
+			}
```

Comment:
As discussed in WhatsApp, data shouldn't be set to easily when editing
First, you need to make sure no one is trying to change the ID
Second, some attributes might be null, as a user can decide to edit only few properties of the recipe, this set would overwrite the existing data in case some properties are null
Third, a user might want to edit the ingredients array and not entirely replace it, you should support merging the data coming from the user with the one already in the database. Check firestore documentation on how to merge data and edit data, maybe @SagivAsraf can help you here, since he's doing a similar thing in the UserService